src/components/PureImagination.js

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react'
+
+export default class extends Component {
+  codeword = 'pure imagination'
+
+  state = { progress: 0 }
+
+  componentDidMount() {
+    window.document.onkeypress = e => {
+      let shouldProgress = e.key == this.codeword[this.state.progress]
+      this.setState({ progress: shouldProgress ? this.state.progress + 1 : 0 })
```

Comment:
```suggestion
      this.setState(state => ({ progress: shouldProgress ? state.progress + 1 : 0 }))
```
Using this.state inside setState can lead to incorrect/different data.